common/web/keyboard-processor/src/keyboards/activeLayout.ts

```diff
@@ -286,13 +302,29 @@ namespace com.keyman.keyboards {
 
     }
 
+    static sanitize(rawRow: LayoutRow) {
+      for(const key of rawRow.key) {
+        // Test for a trailing comma included in spec, added as null object by IE
+        // It has only ever appeared at the end of a row's spec.
+        if(key == null) {
+          rawRow.key.length = rawRow.key.length-1;
+        } else {
+          ActiveKey.sanitize(key);
+        }
+      }
+
+      if(typeof rawRow.id == 'string') {
+        rawRow.id = Number.parseInt(rawRow.id, 10);
+      }
```

Comment:
FWIW, there are only 4 keyboards with stringified row IDs in the keyboards repo in their touch layout files. Compiler normalizes all the others to stringified row IDs. Oh the joys of file compat!